csrc/device_lower/analysis/fused_reduction.cpp

```diff
@@ -104,9 +107,10 @@ class FusionInspector : private IterVisitor {
 
       return reduction_count == 1 && has_valid_tidx_reduction;
     };
+    bool is_cluster_reduction = out->domain()->hasClusterReduction();
```

Comment:
IIRC, this analysis is for translating a reduction + broadcast to an allreduce. Does cluster reduction also follow the same path? Or are you just using this pass to count the number of cluster reductions?

Reply:
It does both. The primary function is to translate a reduction + broadcast to an allreduce. The side job is to count number of cluster reductions.